### ISSUE TITLE
navigation.indexes incompatible with navigation.sections

### ISSUE THREAD

**Issue by worldtiki** [squidfunk/mkdocs-material#3070]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [x] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

### Description

The documentation says the `navigation.indexes` feature [is compatible with all other feature flags](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages) however it doesn't seem to work if I enable navigation.sections.

When only `navigation.indexes` is enabled this feature works ok.
However when combined with `navigation.sections` then the section indices are no longer clickable.

<img width="1414" alt="nobug" src="https://user-images.githubusercontent.com/1187103/135643828-7ba13210-92dc-4ac1-9797-860a9ec77792.png">

<img width="1370" alt="bug" src="https://user-images.githubusercontent.com/1187103/135643838-e8ae97d5-7113-40b6-a4d2-129679c500af.png">

### Expected behaviour

Both features should be compatible with each other.

### Actual behaviour

Section indices are not clickable when both features are enabled.

### Steps to reproduce

Enable both `navigation.sections` and `navigation.indexes`.

```
theme
  features:
    - tabs
    - navigation.tabs
    #- navigation.sections
    - navigation.indexes
```

### Package versions

Using docker image: squidfunk/mkdocs-material:7.3.0

### Configuration

```yaml
site_name: My website
docs_dir: 'src'
site_dir: 'docs'

theme:
  name: material
  icon:
    logo: octicons/rocket-16
  favicon: 'images/favicon.ico'
  features:
    - tabs
    - navigation.tabs
    - navigation.sections
    - navigation.indexes
    - navigation.top
    - instant
    - navigation.tracking
  palette:
    primary: indigo
    accent: blue

plugins:
  - search:
     lang:
        - en

nav:
  - Potato:
      - Foo:
        - 'foo/index.md'
        - Something: 'foo/another_page'
      - Bar:
        - 'bar/index.md'
        - Something: 'bar/another_page'
  - Tomato:
      - Radars: 'something/something.md'

# Repository
repo_name: myrepo
repo_url: https://github.com/myrepo/
edit_uri: edit/master/src/
```

### System information

- Operating system: macOS Big Sur
- Browser: Google Chrome latest

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in 12666a3c003a5dde0987a2d70dbfa7c2b94f7c86.

Additionally, I took the time to restructure the navigation CSS and move to a flex-based approach, as it's now widely supported and seems to be more resilient when it comes to supporting all possible combinations of navigation features. I hope that this doesn't introduce any further regressions and have tested all thinkable combinations, including RTL languages to be sure. If it does, please open new issues. Navigation is a very, very complex beast after all.

Further improvements

- When using `navigation.tabs`, top-level index pages were also not clickable before. This is now fixed.
- Navigation index links will now always stretch to the available space. This means that when clicking into the whitespace between the link and chevron of an index page, the link is followed. Previously, nothing happened.
- Some spacing adjustments related to navigation and source facts (bycatch) in 2d3354bd0e3503c84684c8efc19d02629c733288

**Comment by worldtiki** [comment]
Amazing! Thank you 😊

**Comment by squidfunk** [comment]
Released as part of 7.3.1.
